model_zoo/census_model_sqlflow/wide_and_deep/transform_ops.py

```diff
@@ -0,0 +1,74 @@
+from enum import Enum
+
+
+class TransformOpType(Enum):
+    HASH = (1,)
+    BUCKETIZE = (2,)
+    LOOKUP = (3,)
+    EMBEDDING = (4,)
+    CONCAT = (5,)
+    ARRAY = 6
+
+
+class TransformOp(object):
+    def __init__(self, name, input, output):
+        self.name = name
+        # The input name or a list of input names
+        self.input = input
+        # The output name
+        self.output = output
+        # The type of the TransformOp.
+        # The value is one of `TransformOpType`.
+        self.op_type = None
+
+
+class Vocabularize(TransformOp):
+    def __init__(
+        self, name, input, output, vocabulary_list=None, vocabulary_file=None
+    ):
+        super(Vocabularize, self).__init__(name, input, output)
+        self.op_type = TransformOpType.LOOKUP
+        self.vocabulary_list = vocabulary_list
+        self.vocabulary_file = vocabulary_file
+
+
+class Bucketize(TransformOp):
+    def __init__(self, name, input, output, num_buckets=None, boundaries=None):
```

Comment:
`boundaries` may be enough an we can infer the `num_buckets` according to `boundaries`.

Reply:
My original opinion is that each TransformOP contains the fields from different source
1. User provide in the SQLFlow `COLUMN` expression. User may provide `num_buckets` in the SQLFlow statement.
2. Parameters we infer from data analysis
3. Parameters by walking through the Transform Graph.

Reply:
Yeah. I also think how users utilize those TransformOPs to develop their models. So, it need to make it user-friendly.

Reply:
I see, we can add `required` keyword for the necessary parameters.
What's more, the TransformOp in this python file is samples describing the data structure which has the same field in GO. For E2E ML on SQLFlow, there is no code containing the `TransformOp` in python file.